parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -412,6 +446,41 @@ public byte[] read(byte[] ignored) {
     }
   }
 
+  private static class TimestampInt96Reader extends UnboxedReader<Long> {
+
+    private TimestampInt96Reader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public Long read(Long ignored) {
+      return readLong();
+    }
+
+    @Override
+    public long readLong() {
+      final ByteBuffer byteBuffer =
+          column.nextBinary().toByteBuffer().order(ByteOrder.LITTLE_ENDIAN);
+      return ParquetUtil.extractTimestampInt96(byteBuffer);
+    }
+  }
+
+  private static class TimestampMillisReader extends UnboxedReader<Long> {
+    private TimestampMillisReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public Long read(Long ignored) {
+      return readLong();
+    }
+
+    @Override
+    public long readLong() {
+      return 1000L * column.nextInteger();
```

Comment:
This is valid for `time` but not for `timestamp`. I may have mixed up the `timestamp` reader and `time` reader in an earlier comment. This needs to be `nextLong`.

Reply:
I think the confusion was from this comment: https://github.com/apache/iceberg/pull/11904#discussion_r1906138972

I was talking about the time type, but the code I pasted had the wrong class name, `TimestampMillisReader` should have been `TimeMillisReader`. Timestamps (millis) should use `nextLong` and time (millis) should use `nextInteger`.

Reply:
Fixed in https://github.com/ajantha-bhat/iceberg/pull/74

Reply:
Lack of test coverage in the base code for these milliseconds time, timestamp and int96 timestamps is the reason for back and forth. Tests would have caught this. Will try to add in a follow up.